src/model/simulator.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from cobra.exceptions import OptimizationError
+from cobra.flux_analysis import flux_variability_analysis, pfba
+from cobra.flux_analysis.moma import add_moma
+from cobra.io.dict import model_to_dict
+
+from model import constants, storage
+from model.operations import is_dummy, phase_plane_to_dict
+
+
+logger = logging.getLogger(__name__)
+
+METHODS = [
+    'fba',
+    'pfba',
+    'fva',
+    'pfba-fva',
+    'moma',
+    'lmoma',
+]
+
+
+def simulate(model, method, objective_id, objective_direction, tmy_objectives, to_return):
+    if method not in METHODS:
+        raise ValueError(f"Unsupported simulation method '{method}'")
+
+    with model:
+        if objective_id:
+            model.objective = model.reactions.get_by_id(objective_id)
+        if objective_direction:
+            model.objective.direction = objective_direction
+
+        try:
+            if method == 'fba':
+                solution = model.optimize()
+            elif method == 'pfba':
+                solution = pfba(model)
+            elif method == 'fva':
+                # FIXME: accept list of relevant fva reactions to calculate
+                solution = flux_variability_analysis(model)
+            elif method == 'pfba-fva':
+                # FIXME: accept list of relevant fva reactions to calculate
+                solution = flux_variability_analysis(model, fraction_of_optimum=1, pfba_factor=1.05)
+            elif method in ('moma', 'lmoma'):
```

Comment:
I think that keeping MOMA around (and tested) while we're not using the method is an unnecessary burden.

Reply:
Yes, it is. I'll remove it for now.